### DISCUSSION TITLE
Combining CoreFreq with Active AMD Pstate

### DISCUSSION THREAD

**Discussion by LethalManBoob** [cyring/CoreFreq#432] [Q&A]
Is this possible? Configuration in the pic below. Also if you have any recommended settings please let me know. Thanks.
![image](https://user-images.githubusercontent.com/30273211/231558848-083a5fa7-1ccb-4906-827d-f60d7f8501c1.png)

**Comment by cyring** [comment]
Now what you can do:

##  C-States
* Clock Source = corefreq_tsc (optionally)
* Idle Limit:
  * below than `C2` for energy savings
  * `POLL` for lowest latency
* C-State method = `I/O` or `Halt` or `MWAIT` instruction
  Follow the method impacts in the "Sensors" view.

![2023-04-12-220817_644x1012_scrot](https://user-images.githubusercontent.com/11563789/231572498-520a34a2-fa60-4767-a8ad-5b7aec382709.png)

## P-States
1. Disable CPB (Core Performance Boost)
2. Change the P-State `1C` (single Core) and/or `2C`
3. Re-enable CPB
4. Change the Target frequency

![2023-04-12-221427_644x1012_scrot](https://user-images.githubusercontent.com/11563789/231574252-25b66f59-3e5e-4ca9-9df9-12dd50f57e67.png)

![2023-04-12-221517_644x1012_scrot](https://user-images.githubusercontent.com/11563789/231574271-d077f0c2-3b51-4f0f-8e56-0e114d298e37.png)

* To check results, it is best to switch to `Absolute frequency` mode. Press `!` shortcut.

![2023-04-12-222255_643x73_scrot](https://user-images.githubusercontent.com/11563789/231575586-c7930a02-f0a7-4f24-9f2e-0bce13a3ef2c.png)

## Frequency Max
* Frequency Max is really **experimental**.
Somehow it impacts Base Clock.
**Save your files**

1. Choose `hpet` as the Clock Source
2. In "Technologies", disable `CPB`
3. In "Processor", select "Max" in "Frequency" section
4. CPU selector will open, choose "Processor" to alter all of them
5. "Max Clock Ratio" will open, change to a desired frequency 

![2023-04-12-222847_644x1012_scrot](https://user-images.githubusercontent.com/11563789/231577623-62ec19b4-e44f-4275-bf1b-6270d0fdd004.png)

**Comment by LethalManBoob** [comment]
so is that a yes... or?

**Comment by cyring** [comment]
> so is that a yes... or?

I would say yes.
But can you rephrase your question ?

**Comment by LethalManBoob** [comment]
Is there a way to enable AMD-Pstate EPP and allow it to continue controlling what it does but alongside corefreq controlling everything else?

**Comment by cyring** [comment]
> Is there a way to use amd pstate epp for controlling what it does (e.g govornors and clockspeeds) but have corefreq control everyting else

Does not handle the EPP driver by AMD

But can manage the EPP hint when CPPC is BIOS enabled

![2023-04-13-002604_492x418_scrot](https://user-images.githubusercontent.com/11563789/231599427-fac3aa8f-be7b-4425-9451-22d1c657eb7d.png)

**Comment by cyring** [comment]
![image](https://user-images.githubusercontent.com/11563789/231601900-62975342-26a5-4e33-a169-b44b104b7afe.png)
